models.py

```diff
@@ -232,9 +232,37 @@ def location_network(rnn_state, sampling, std=params['std'],
     # core network
     # TODO: parameterize rnn_cell, i.e. implement split cell from paper
     with tf.variable_scope('core_network', reuse=tf.AUTO_REUSE):
-        rnn_cell = tf.nn.rnn_cell.LSTMCell(params['core_size'])
-        state = rnn_cell.zero_state(batch_size, tf.float32)
-
+        if params['core_type'] == 'LSTM':
+            rnn_cell = tf.nn.rnn_cell.LSTMCell(params['core_size'])
+            state = rnn_cell.zero_state(batch_size, tf.float32)
+
+        else:
+            def rnn_cell(glimpse, state): 
+                
+                dense_state = tf.layers.dense(
+                    inputs=state,
+                    units=params['core_size'],
+                    activation=None)
+            
+                dense_glimpse = tf.layers.dense(
+                    inputs=glimpse,
+                    units=params['glimpse_out_size'],
+                    activation=None)
+                
+                dstate_plus_dglimpse = tf.add(
+                        x=dense_state,
+                        y=dense_glimpse)
+                
+                rnn_cell = tf.contrib.rnn.BasicRNNCell(
+                    num_units=params['core_size'],
+                    activation=tf.nn.relu)
+    
+                output, state = tf.dynamic_rnn(
+                        cell=rnn_cell,
+                        inputs=dstate_plus_dglimpse)
+                
+                return output, state
```

Comment:
By convention, `rnn_cell`s like yours are supposed to return an output and a state.  But this simple `rnn` does not maintain a state the way that an LSTM does.  So, the thing to do here is kind of strange, but: `return output, output`